### PR TITLE
3D vertex reconstruction

### PR DESCRIPTION
**Pull request by christophwelling**
Adds a module for a full 3D neutrino vertex reconstruction as well as a documentation on how to use the vertex reconstruction modules.

### CODE REVIEW

File: NuRadioReco/autodoc/pages/how_to/vertex_reconstruction.rst [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+NuRadioReco provides a script to produce the lookup tables  at
+`NuRadioReco/modules/neutrinoVertexReconstructor/create_lookup_tables.py`
+The default settings should work well, just pay attention to some points:
+
+  - You need to calculate lookup tables for all antenna depths you want to use in the
+    reconstruction
+  - Make sure the maximum depth and horitontal distance are large enough. They define
+    the region where you can search for the vertex
+  - Make sure you use the right ice model
```
Comment (lpyras) [inline NuRadioReco/autodoc/pages/how_to/vertex_reconstruction.rst:32]: :)

File: NuRadioReco/autodoc/pages/how_to/vertex_reconstruction.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+The neutrino2DVertexReconstructor uses antennas that are on the same string and takes advantage of
+resulting symmetry to speed up the reconstruction. Thanks to the rotational symmetry around the
+z-axis, the problem can be reduced to two dimensions and the vertex position is described by the
+horizontal distance from the center and the depth. The disadvantage is that only channels on the same
+detector string can be used and that deviations from the cylindrical symmetry by a real detector due
+to some uncertainty in the exactl antenna positions may influence the resulting accuracy.
+It can also not reconstruct the azimuth coordinate of the vertex position.
+
+The neutrino3DVertexReconstructor can utilize all all detector channels of the same antenna type and
```
Comment (lpyras) [inline NuRadioReco/autodoc/pages/how_to/vertex_reconstruction.rst:95]: double word "all all"

File: NuRadioReco/modules/neutrinoVertexReconstructor/create_lookup_table.py [hunk 3]
```diff
@@ -49,15 +49,21 @@
...
 )
+parser.add_argument(
+    '--ice_model',
+    type=str,
+    default='greenland_simple',
+    help='Name of the ice model to be used'
+)
 args = parser.parse_args()
-x_pos = np.arange(-args.r_max, -args.r_min, args.d_r)
+x_pos = np.arange(args.r_min, args.r_max, args.d_r)
```
Comment (lpyras) [inline NuRadioReco/modules/neutrinoVertexReconstructor/create_lookup_table.py:62]: variable names have are called x and r, but thats probably okay.

File: NuRadioReco/modules/neutrinoVertexReconstructor/create_lookup_table.py [hunk 4]
```diff
@@ -66,8 +72,8 @@
 
 lookup_table = {
     'header': {
-        'x_min': -args.r_min,
-        'x_max': -args.r_max,
+        'x_min': args.r_min,
+        'x_max': args.r_max,
```
Comment (lpyras) [inline NuRadioReco/modules/neutrinoVertexReconstructor/create_lookup_table.py:76]: same here

**Review by lpyras — APPROVED** [review]
Looks good, I like the debug plots.

File: NuRadioReco/autodoc/pages/how_to/vertex_reconstruction.rst (hunk 1 above)
Comment (christophwelling) [reply NuRadioReco/autodoc/pages/how_to/vertex_reconstruction.rst:32]: Not sure how to reply to this comment:- ;-)

**Review by shallmann — APPROVED** [review]
Ok with me. @lpyras what was your worry on parameters x, r?

**Comment by lpyras** [comment]
> Ok with me. @lpyras what was your worry on parameters x, r?

Just that x_min is defined by r_min. But nothing serious.

**Comment by shallmann** [comment]
Ah, ja, right... I leave the decision to Christoph if he feels like renaming the x_*, i_x, xx to r_*, i_r, rr...

**Review by cg-laser — APPROVED** [review]
I didn't have a look at the module, but because it is an isolated reconstruction module, it is save to merge.

**Comment by anelles** [comment]
> I didn't have a look at the module, but because it is an isolated reconstruction module, it is save to merge.

I agree that it is save to merge, but just for the sake of code quality, I think we should also review analysis modules. And if only to keep people on their toes about writing legible code.

**Pull request merged**